src/components/AddImagePopup/AddImagePopup.vue

```diff
@@ -99,14 +111,17 @@ export default class AddImagePopup extends Vue {
     private deleteImageBySrc!: (src: string) => void;
     // gallery module functionality
 
-    private uploadImages(keyWord: string): void {
+    private async uploadImages(keyWord: string): Promise<void> {
+      this.clearImages();
+
       if (!keyWord) {
         alert('The search string shouldn`t be empty');
         return;
       }
-
-      this.uploadImagesAction(keyWord);
       this.showDisclaimer = false;
+      this.showLoader = true;
+
+      this.showLoader = !await this.uploadImagesAction(keyWord);
```

Comment:
use       this.showLoader = true; or this.showLoader = !await this.uploadImagesAction(keyWord);. Why you use both?